fw/hpack.c

```diff
@@ -1253,13 +1261,47 @@ tfw_hpack_hdr_set(TfwHPack *__restrict hp, TfwHttpReq *__restrict req,
 		parser->_hdr_tag = TFW_HTTP_HDR_RAW;
 		break;
 	default:
-		WARN_ON_ONCE(1);
+		T_WARN("%s: HTTP/2 request dropped: unexpected HPACK entry tag"
+			" = %d\n", __func__, entry->tag);
 		return T_DROP;
 	}
 
 	return T_OK;
 }
 
+static bool
+tfw_hpack_entry_tag_valid(unsigned int entry_tag)
+{
+	bool is_valid = false;
+
+	switch(entry_tag) {
+	case TFW_TAG_HDR_H2_METHOD:
+	case TFW_TAG_HDR_H2_SCHEME:
+	case TFW_TAG_HDR_H2_AUTHORITY:
+	case TFW_TAG_HDR_H2_PATH:
+	case TFW_TAG_HDR_ACCEPT:
+	case TFW_TAG_HDR_AUTHORIZATION:
+	case TFW_TAG_HDR_CACHE_CONTROL:
+	case TFW_TAG_HDR_CONTENT_LENGTH:
+	case TFW_TAG_HDR_CONTENT_TYPE:
+	case TFW_TAG_HDR_COOKIE:
+	case TFW_TAG_HDR_HOST:
+	case TFW_TAG_HDR_IF_MODIFIED_SINCE:
+	case TFW_TAG_HDR_IF_NONE_MATCH:
+	case TFW_TAG_HDR_PRAGMA:
+	case TFW_TAG_HDR_REFERER:
+	case TFW_TAG_HDR_X_FORWARDED_FOR:
+	case TFW_TAG_HDR_USER_AGENT:
+	case TFW_TAG_HDR_RAW:
+		is_valid = true;
+		break;
+	default:
+		T_WARN("%s: HTTP/2 request dropped: invalid HPACK entry tag"
+			" = %d\n", __func__, entry_tag);
```

Comment:
I think this should be a debug message: we print warnings to users and users may struggle on watching function names. Also the wild Internet traffic is full of bots scanning IP ranges in security vulnerabilities, so we might see a lot of such messages.